packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -196,14 +197,16 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
     }
 
     async $start(params: PluginManagerStartParams): Promise<void> {
+        this.configStorage = params.configStorage;
```

Comment:
hi, why do we have breaking changes in plugin-manager for the tests ?

Reply:
I wonder how it can be customized. Do you patch source code?

Reply:
We need a function for test to activate a plugin eagerly: `PkuginManager.$activatePlugin`. While running tests it's turned out that plugin activation was bogus in some parts:
- different instances of `configStorage` are used for each start, so if the workspace path is changed previous starts are still using old path, `configStorage` was turned into the property to fix it
- activation reporting started too eagerly for each activation event instead of only once when the plugin actually getting activated

The code was reorganized to fix it. I believe the single breaking change is removing `configStorage` parameter from `registerPlugin` method. Others are not breaking, a new method and changes fo control flow for activation reporting.

Reply:
> The code was reorganized to fix it. I believe the single breaking change is removing configStorage parameter from registerPlugin method. Others are not breaking, a new method and changes fo control flow for activation reporting.

We've a custom plugin-manager (extends the current-one) and then registerPlugin signature is now changed

Reply:
ok, I will add to CHANGELOG

Reply:
@benoitf I just wanted to check one thing since we discuss it. It seems that `configStorage`  is not different from a plugin to plugin, but should be configured for each manager and updated for all plugins. Would not it better if we pass it via `$init` as other manager configurations and refresh with `$update`? Do I overlook something?

Reply:
I think it should be fine as well (just breaking again API)

Reply:
I will file a separate issue. I don't think we handle workspace changes correctly. VS Code will start stop and start all managers, but we are patching workspace storage paths on the fly. It is error prone.

Reply:
https://github.com/eclipse-theia/theia/issues/7472

Reply:
updated CHANGELOG